src/ucs/sys/event_set.h

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (C) Hiroyuki Sato. 2019. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_EVENT_SET_H
+#define UCS_EVENT_SET_H
+
+#include <ucs/type/status.h>
+
+/**
+ * ucs_sys_event_set structure used in ucs_event_set_XXX functions.
+ *
+ */
+typedef struct ucs_sys_event_set ucs_sys_event_set_t;
+
+
+typedef struct ucs_sys_event_set_data {
+    int fd;      /* event */
+    int events;  /* event_set events */
+} ucs_sys_event_set_data_t;
+
+
+/**
+ * ucs_event_set_handler call this handler for notifying event
+ *
+ */
+typedef ucs_status_t (*event_set_handler_t)(int nready,
```

Comment:
could be simpler API:
`typedef ucs_status_t (*event_set_handler_t)(int fd, int events)`
the callback will handle 1 event every time